src/happy.rs

```diff
@@ -0,0 +1,31 @@
+use std::str::FromStr;
+
+use hickory_client::{
+    client::{Client, SyncClient},
+    error::ClientResult,
+    op::DnsResponse,
+    rr::{DNSClass, Name, RecordType},
+    udp::UdpClientConnection,
+};
+
+/*
+TODO:
+This needs to be TCP and have Timeout Retries
+
+*/
+pub fn search(identifier: Name, rtype: RecordType) -> anyhow::Result<DnsResponse> {
+    debug!("searching {:?} for {}", rtype, identifier);
+    let socket_addr = std::net::SocketAddr::new(std::net::IpAddr::from_str("1.1.1.1")?, 53);
+    let client = happy_eyeballs::connect(socket_addr)?;
+    let questionable_addr = client.peer_addr()?;
+    let conn = UdpClientConnection::new(socket_addr).unwrap();
```

Comment:
No unwrapping! We should probably enable the https://rust-lang.github.io/rust-clippy/master/index.html#/unwrap_used and https://rust-lang.github.io/rust-clippy/master/index.html#/expect_used lints to let clippy remind us of doing proper error handling here.